File: app/src/main/java/pl/coddev/applu/c/Log.java
package pl.coddev.applu.c;

import com.crashlytics.android.Crashlytics;

/**
 * Created by Piotr Woszczek on 27/04/15.
 */
public class Log {


    public static void d(String tag, String message){
        logIt(android.util.Log.DEBUG, tag, message);
    }

    public static void i(String tag, String message){
        logIt(android.util.Log.INFO, tag, message);
    }

    public static void e(String tag, String message){
        logIt(android.util.Log.ERROR, tag, message);
    }

    public static void v(String tag, String message){
        logIt(android.util.Log.VERBOSE, tag, message);
    }

    public static void w(String tag, String message){
        logIt(android.util.Log.WARN, tag, message);
    }

    private static void logIt(int level, String tag, String message){
        if(Constants.CRASHLYTICS_LOGS) {
            Crashlytics.log(level, tag, message);
        } else {
            android.util.Log.println(level, tag, message);
        }
    }
}

File: app/src/main/java/pl/coddev/applu/utils/Log.kt
package pl.coddev.applu.utils

import android.util.Log

/**
 * Created by Piotr Woszczek on 27/04/15.
 */
object Log {
    @JvmStatic
    fun d(tag: String, message: String) {
        logIt(Log.DEBUG, tag, message)
    }

    fun i(tag: String, message: String) {
        logIt(Log.INFO, tag, message)
    }

    @JvmStatic
    fun e(tag: String, message: String) {
        logIt(Log.ERROR, tag, message)
    }

    fun v(tag: String, message: String) {
        logIt(Log.VERBOSE, tag, message)
    }

    fun w(tag: String, message: String) {
        logIt(Log.WARN, tag, message)
    }

    private fun logIt(level: Int, tag: String, message: String) {
        if (Constants.CRASHLYTICS_LOGS) {
            //Crashlytics.log(level, tag, message);
        } else {
            Log.println(level, tag, message)
        }
    }
}